dist/tools/compile_test/compile_like_murdock.py

```diff
@@ -0,0 +1,231 @@
+    'avr-rss2',
+    'avsextrem',
+    'bastwan',
+    'bluepill',
+    'b-l475e-iot01a',
+    'cc1352-launchpad',
+    'cc2650-launchpad',
+    'derfmega128',
+    'dwm1001',
+    'esp32-ci',
+    'esp32-heltec-lora32-v2',
+    'esp32-olimex-evb',
+    'esp32c3-ci',
+    'esp32c3-devkit',
+    'esp32s3-ci',
+    'esp32s3-devkit',
+    'esp32s2-ci',
+    'esp32s2-devkit',
+    'esp8266-ci',
+    'esp8266-esp-12x',
+    'hamilton',
+    'hifive1',
+    'mbed_lpc1768',
+    'mega-xplained',
+    'microbit',
+    'mulle',
+    'native',
+    'nrf52840dk',
+    'nrf9160dk',
+    'nucleo-f072rb',
+    'nucleo-f103rb',
+    'nucleo-f207zg',
+    'nucleo-f334r8',
+    'nucleo-f429zi',
+    'nucleo-f767zi',
+    'nucleo-g071rb',
+    'nucleo-g474re',
+    'nucleo-l011k4',
+    'nucleo-l073rz',
+    'nucleo-l152re',
+    'nucleo-l433rc',
+    'nucleo-l552ze-q',
+    'pba-d-01-kw2x',
+    'p-nucleo-wb55',
+    'qn9080dk',
+    'remote-revb',
+    'same54-xpro',
+    'samr21-xpro',
+    'seeedstudio-gd32',
+    'slstk3400a',
+    'sltb001a',
+    'slwstk6220a',
+    'stm32f723e-disco',
+    'stm32mp157c-dk2',
+    'waspmote-pro',
+    'waveshare-nrf52840-eval-kit',
+    'weact-f401ce',
+    'z1'
+]
+
+
+def _end(sec, job):
+    time_str = str(datetime.timedelta(seconds=sec))
+    thing, ratio = random.choice(_RATIOS)
+    val = ratio * sec * job
+    print(f"This took {time_str}...")
+    print(f"You could have {val} {thing}")
+
+
+def _all_apps(cwd):
+    cmd = ['make', 'info-applications', '--no-print-directory']
+    out = subprocess.check_output(cmd, cwd=cwd)
+    return out.decode("utf-8", errors="replace").split()
+
+
+def _supported_boards(boards, env, cwd):
+    cmd = ['make', 'info-boards-supported', '--no-print-directory']
+    out = subprocess.check_output(cmd, env=env, cwd=cwd)
+    supported_boards = out.decode("utf-8", errors="replace").split()
+    return [brd for brd in supported_boards if brd in boards]
+
+
+def _supported_boards_from_cpu(cpu, env, cwd):
+    cmd = (f'FEATURES_REQUIRED=cpu_{cpu} make info-boards-supported '
+           '--no-print-directory')
+    out = subprocess.check_output(cmd, shell=True, env=env, cwd=cwd)
+    return out.decode("utf-8", errors="replace").split()
+
+
+def _build(app, board, jobs, env, cwd):
+    cmd = (f'/bin/bash -c "source .murdock; JOBS={jobs} '
+           f'compile {app} {board}:gnu"')
+    try:
+        out = subprocess.check_output(cmd, env=env, shell=True,
+                                        cwd=cwd, stderr=subprocess.STDOUT)
+        out = out.decode("utf-8", errors="replace")
+        print(f"{app: <30} {board: <30} PASS")
+    except subprocess.CalledProcessError as err:
+        err.output = err.output.decode("utf-8", errors="replace")
+        lines = err.output.split("\n")
+        if "mismatch" in err.output:
+            print(f"{app: <30} {board: <30} FAIL: Kconfig hash mismatch")
+        elif lines[-3].startswith('< ') or lines[-3].startswith('> '):
```

Comment:
Maybe these should be switched as module mismatches almost always means hash mismatches but not the other way around.